torch/csrc/distributed/spmd/event_handler_impl.cpp

```diff
@@ -0,0 +1,209 @@
+#include <torch/csrc/distributed/spmd/event_handler_impl.h>
+
+#include <torch/csrc/autograd/engine.h>
+#include <torch/csrc/autograd/functions/accumulate_grad.h>
+#include <torch/csrc/autograd/utils/lambda_post_hook.h>
+
+
+namespace torch {
+namespace distributed {
+namespace spmd {
+
+using c10::IValue;
+
+namespace {
+
+template <class T>
+std::vector<std::shared_ptr<Future>> createOneFutureEvent(
+    c10::intrusive_ptr<T> event) {
+  auto future = std::make_shared<Future>(at::AnyClassType::get());
+  future->markCompleted(
+      IValue(c10::static_intrusive_pointer_cast<Event>(event)));
+  std::vector<std::shared_ptr<Future>> futures;
+  futures.reserve(1);
+  futures.emplace_back(std::move(future));
```

Comment:
nit: Can't we do something like this:

```
return std::vector<std::shared_ptr<Future>> {future};
```

Reply:
I think this is a pure performance optimization by avoiding the **atomic** increment caused by copying a shared pointer.